packages/sanity/src/core/form/inputs/files/common/PlaceholderText.tsx

```diff
@@ -34,27 +37,27 @@ export function PlaceholderText(props: Props) {
   }, [directUploads, hoveringFiles, isFileType, readOnly, rejectedFilesCount])
 
   const messageText = useMemo(() => {
-    let message = `Drag or paste ${type} here`
-
     if (!directUploads) {
-      return `Can't upload files here`
+      return t('inputs.files.common.placeholder.upload-not-supported')
     }
 
     if (readOnly) {
-      message = 'Read only'
+      return t('inputs.files.common.placeholder.read-only')
     }
 
     if (hoveringFiles && directUploads && !readOnly) {
       if (acceptedFiles.length > 0) {
-        message = `Drop to upload ${type}`
+        return t('inputs.files.common.placeholder.drop-to-upload', {type})
       }
       if (rejectedFilesCount > 0) {
-        message = `Can't upload ${rejectedFilesCount} file${rejectedFilesCount > 1 ? 's' : ''} here`
+        return t('inputs.files.common.placeholder.cannot-upload-some-files', {
+          count: rejectedFilesCount,
+        })
       }
     }
 
-    return message
-  }, [acceptedFiles.length, directUploads, hoveringFiles, readOnly, rejectedFilesCount, type])
+    return t('inputs.files.common.placeholder.drag-or-paste-to-upload')
```

Comment:
The translations refer to `{{type}}` which is not passed here. Either rewrite the translation strings to not include the type, or pass the `context` option to allow translating the individual asset types.